### PR TITLE
Various SCA and Coverity fixes

### PR DESCRIPTION
**Pull request by dalgaaf**
Fixed 40 Coverity issues and some other findings from cppcheck. 

Fixed type of Coverity issues: UNUSED_VALUE, MISSING_BREAK, UNREACHABLE, IDENTICAL_BRANCHES, UNINTENDED_INTEGER_DIVISION, OVERRUN, REVERSE_INULL, NEGATIVE_RETURNS, SWAPPED_ARGUMENTS, OVERFLOW_BEFORE_WIDEN.

Check especially the SWAPPED_ARGUMENTS issues.

### CODE REVIEW

**Comment by tchaikov** [comment]
just reviewed the patches. some of them seem like critical fixes in rbd. and we might want to have an alternative fix in  4fe2c08 .

**Comment by tchaikov** [comment]
@dalgaaf this pull request does not merge anymore.

**Comment by dalgaaf** [comment]
@tchaikov I know, I work on an update

**Comment by dalgaaf** [comment]
Rebased and fixed issues from the comments.

**Comment by tchaikov** [comment]
the failure is noteworthy.

**Comment by dalgaaf** [comment]
@tchaikov I tried to bisect the isse but unfortunately it always points to another patch. Have to take a deeper look at it.

**Comment by tchaikov** [comment]
yup. i reproduced this error locally. will also give it a stab later today.

File: ceph-object-corpus [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit 7bd9bbe235c59527dd1ab192fed21944da686d20
+Subproject commit da43eb119558fb46d86e62f3b82d5ca63546f69e
```
Comment (tchaikov) [inline ceph-object-corpus]: thanks to @dachary , we need to remove this commit from this pull request. and it will be good then =D

Comment (dalgaaf) [inline ceph-object-corpus]: jepp ... commit -sva is sometimes bad ;-)

**Comment by tchaikov** [comment]
`Reviewed-by: Kefu Chai <kchai@redhat.com>` after the qa run

**Comment by dalgaaf** [comment]
Fixed merge with master.

**Comment by dalgaaf** [comment]
Accidently closed in comment

**Comment by dalgaaf** [comment]
Is QA and testing finished?

**Comment by tchaikov** [comment]
@dalgaaf, it failed with a test: http://pulpito.ceph.com/ubuntu-2015-07-28_00:57:07-rados-wip-kefu-t3sting---basic-multi/

i am still looking at it.

**Comment by tchaikov** [comment]
the performed rados qa run tests are scatter in sepia and my teuthology test bed

- http://149.202.175.39:8081/ubuntu-2015-07-31_09:20:30-rados-wip-kefu-t3sting---basic-openstack/

some of the test failure were fixed by #4958 and #5395 which are fixed in master. and some were due to environmental problems.

**Pull request merged**